.buildkite/pipeline.yml

```diff
@@ -13,70 +13,70 @@ steps:
   # Build the app
   #################
   - label: ":pipeline: Build"
-    key: "build"
-    command: ".buildkite/commands/build-for-testing.sh"
+    key: build
+    command: .buildkite/commands/build-for-testing.sh
     plugins: [$CI_TOOLKIT]
     notify:
       - github_commit_status:
-          context: "Build"
+          context: Build
 
   #################
   # Create Prototype Build
   #################
-  - label: "🛠 Prototype Build"
-    command: ".buildkite/commands/prototype-build.sh"
+  - label: 🛠 Prototype Build
```

Comment:
I'm not sure why (just a gut feeling, I guess?) but I feel like it would make sense to keep the quotes for `label` here…

I think I'm feeling that way probably because:
 - The fact that the string starts with an emoji makes it feel like a special case where quoting wouldn't hurt
 - The `label` attribute of other steps are already quoted, especially the ones starting with `:` to introduce a non-Unicode, Buildkite-specific `:emoji:` code… and thus at that point since some `label` entries need quoting, maybe it'd feel more consistent to quote them for all steps?

Totally opinionated, and that's bike shedding at this point, so this is just my 2 cents though, and whatever floats the team's boat.

Reply:
I was also on the fence on the ones with unicode emojis. Maybe we can replace the unicode emojis with a Buildkite `:emoji:` so it becomes obvious why we're keeping the quotes for them?

Reply:
+1 for using the `:emoji:` syntax here (and everywhere else).

One additional argument in favor of this is that if we want to use [a custom emoji](https://github.com/buildkite/emojis#emoji-reference), the only way to do it is via `:emoji:`. To maintain consistency, it then makes sense to me to use `:emoji:` for _all_ instances, even those that are standard.

Reply:
Updated on 142cbdbdea893c4c28afac0dbfb3b322f13961b8.

Reply:
The drawback of using `:emoji:` instead of the unicode character is that the generated name for the GitHub commit status will include `emoji-` in the status name. so a label of `🛠 Prototype Build` will make Buidkite generate a GitHub commit status of `prototype-build` but a label of `:hammer_and_wrench: Prototype Build` will make Buildkite generate a GitHub commit status of `hammer-and-wrench-prototype-build`…

I guess this makes this whole discussion go back to the debate of using custom GitHub commit status names for each of our steps explicitly (paaHJt-78h-p2 / paaHJt-4cs-p2)…

---

For the record, I have a small ruby script ready that takes a pipeline YAML file as input and add the `notify: { github_commit_status: { context: … } }` to any step that doesn't have it, but using a name for the `context` being custom-built from the label (don't snake-case it like Buildkite does, keep spaces and only remove emojis and :emoji: codes from the label string…).

It would be as easy to use as `add_missing_github_commit_statuses ${PIPELINE:-pipeline.yml} | buildkite-agent pipeline upload` for the upload step of our pipelines that run on the `upload` agents… but the question is how do we install such a script on the `upload` agents?

 - Since those `upload` agents are EC2 machines in AWS that are based on Buildkite's official AMI, there's no easy way to install custom scripts or hooks on the default agent like we'd do if that was an agent provisioned via an ansible playbook etc). We could build our own custom AMI based on the Buildkite one (like we do for e.g. `android` agents) just to install that script in the AMI, but I don't think it's worth the effort to maintain custom AMIs just for the purpose of adding a single, <20LoC script…
 - We could hijack the S3 secrets (that are installed on each agent before the commands are run) by wrapping this script as a bash function in `.mobile-secrets/CI/secrets/env` (that would call `ruby -e <<SCRIPT … SCRIPT` or something like that, with the Ruby script inline in the bash function). I think that would work. But it feels a bit odd to have such a script in `mobile-secrets` while there's nothing really secret about it
 - We could use the per-instance bootstrap script (`BootstrapScriptUrl` parameter Buildkite's Elastic CI Stack) to install such a script in the agent. Actually, to my understanding, this parameter is intended exactly for such purposes of running some custom script at start of the agent to customize additional stuff. The issue is that this parameter has to point to a S3 object or HTTPS public URL, which then begs the question of where we'd store said bootstrap script that would install our `add_missing_github_commit_statuses` Ruby script into the `$PATH`. We could not just store it in `buildkite-ci` because it's a private repo so wouldn't be accessible. We could store it in S3, but that means that every time we modify this script we'd need to remember to re-deploy it into S3. But maybe that's the lesser of all those evils?

<details><summary>📜 Here's what the script in question looks like</summary>

```ruby
#!/usr/bin/env ruby
# frozen_string_literal: true

require 'yaml'

def clean_label(label)
  label
    # Remove `:emoji:` names
    .gsub(/:[a-z_-]+:/, '')
    # Only keep certain category of unicode characters (Letters, some Marks, Numbers, some Punctuation, Spacing)
    .scan(/\p{L}|\p{Mc}|\p{Me}|\p{N}|\p{Pd}|\p{Ps}|\p{Pe}|\p{Zs}/).join
    # Remove extra spaces at start/end + replace runs of repeated spaces with a single one
    .strip.squeeze(' ')
end

def process(step:)
  # Recurrence on groups
  step['steps']&.each { |s| process(step: s) }

  # Otherwise if normal step with a label, process it
  label = step['label']
  return if label.nil?

  gcs = Array(step['notify']).find { |n| !n['github_commit_status'].nil? }
  return unless gcs.nil? # Already one specified explicitly, exit early

  # Append the custom-generated `notify` entry for GitHub Commit Status
  gcs = { 'github_commit_status' => { 'context' => clean_label(label) } }
  step['notify'] ||= []
  step['notify'].append(gcs)
end

y = YAML.load(ARGF.read)
y['steps'].each { |s| process(step: s) }
puts y.to_yaml
```
</details>

This is a bit frustrating because this `add_missing_github_commit_statuses` script is super simple and would perfectly solve our conundrum here (providing custom statuses but also not render the pipelines too verbose nor risk forgetting to add statuses for each step we add), and the only tricky part is to make it available on our `upload` agents…

---

Anyway, in the meantime, maybe it'd just be simpler to add those `notify: [{github_commit_status: { context: … }}]` entries to all our steps manually one by one, even if that makes the pipelines verbose and the full attribute is a bit of a PITA to write manually…?

Reply:
I've added the missing `github_commit_status` to the main `pipeline.yml` (e35406b121f931af8045fd158c0e745afffbc81d) and created the PR https://github.com/Automattic/buildkite-ci/pull/518, to clean up and standardize the status checks as we've been discussing.